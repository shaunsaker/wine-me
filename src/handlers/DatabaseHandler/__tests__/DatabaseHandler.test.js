import React from 'react';
import renderer from 'react-test-renderer';

import { DatabaseHandler } from '..';

describe('DatabaseHandler', () => {
  const spies = [];
  const dispatch = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle handleSyncData', () => {
      spies[0] = jest.spyOn(DatabaseHandler.prototype, 'syncCategories');
      spies[1] = jest.spyOn(DatabaseHandler.prototype, 'syncFeaturedPlaces');
      spies[2] = jest.spyOn(DatabaseHandler.prototype, 'syncPlaces');
      spies[3] = jest.spyOn(DatabaseHandler.prototype, 'syncSearchAreas');
      spies[4] = jest.spyOn(DatabaseHandler.prototype, 'syncUserCheckIns');
      spies[5] = jest.spyOn(DatabaseHandler.prototype, 'syncUserSettings');

      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.handleSyncData();

      expect(spies[0]).toHaveBeenCalled();
      expect(spies[1]).toHaveBeenCalled();
      expect(spies[2]).toHaveBeenCalled();
      expect(spies[3]).toHaveBeenCalled();
      expect(spies[4]).toHaveBeenCalled();
      expect(spies[5]).toHaveBeenCalled();
    });

    it('should handle syncCategories', () => {
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.syncCategories();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });

    it('should handle syncFeaturedPlaces', () => {
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.syncFeaturedPlaces();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });

    it('should handle syncPlaces', () => {
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.syncPlaces();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });

    it('should handle syncSearchAreas', () => {
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.syncSearchAreas();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });

    it('should handle syncUserCheckIns', () => {
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.syncUserCheckIns();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });

    it('should handle syncUserSettings', () => {
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.syncUserSettings();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });
  });

  describe('lifecycle methods', () => {
    it('should call handleSyncData if authenticated in componentDidMount', () => {
      spies[0] = jest.spyOn(DatabaseHandler.prototype, 'handleSyncData');
      renderer.create(<DatabaseHandler dispatch={dispatch} authenticated />);

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should call handleSyncData on new authenticated in componentDidUpdate', () => {
      spies[0] = jest.spyOn(DatabaseHandler.prototype, 'handleSyncData');
      const component = renderer.create(<DatabaseHandler dispatch={dispatch} />);

      component.update(<DatabaseHandler dispatch={dispatch} authenticated />);

      expect(spies[0]).toHaveBeenCalled();
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
