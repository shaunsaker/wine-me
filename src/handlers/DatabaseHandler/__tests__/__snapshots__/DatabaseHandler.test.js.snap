// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatabaseHandler methods should handle syncCategories 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "SET_CATEGORIES",
          },
          "pathParts": Array [
            "categories",
          ],
        },
        "type": "sync",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DatabaseHandler methods should handle syncFeaturedPlaces 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "payload": Object {
              "ref": "featuredPlaces",
            },
            "type": "SET_FEATURED_PLACES",
          },
          "pathParts": Array [
            "featured_places",
          ],
        },
        "type": "sync",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DatabaseHandler methods should handle syncPlaces 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "SET_PLACES",
          },
          "pathParts": Array [
            "places",
          ],
        },
        "type": "sync",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DatabaseHandler methods should handle syncSearchAreas 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "SET_SEARCH_AREAS",
          },
          "pathParts": Array [
            "search_areas",
          ],
        },
        "type": "sync",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DatabaseHandler methods should handle syncUserCheckIns 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "SET_USER_CHECK_INS",
          },
          "pathParts": Array [
            "check_ins",
          ],
          "query": Array [
            "uid",
            "==",
            undefined,
          ],
        },
        "type": "sync",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DatabaseHandler methods should handle syncUserSettings 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "SET_USER_SETTINGS",
          },
          "pathParts": Array [
            "user_settings",
          ],
          "query": Array [
            "uid",
            "==",
            undefined,
          ],
        },
        "type": "sync",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DatabaseHandler renders renders with minimum required props 1`] = `null`;
